skbio/stats/composition.py

```diff
@@ -608,10 +610,272 @@ def centralize(mat):
+    ...      [20.3, 21.3, 10.2, 10.3, 10.1, 10.6, 10.4]],
+    ...     index=['s1','s2','s3','s4','s5','s6'],
+    ...     columns=['b1','b2','b3','b4','b5','b6','b7'])
+
+    Then create a create a category vector.  In this scenerio, there
+    are only two class, so the first three samples fall under the first
+    class while the last three samples fall under the last class
+    >>> cats = pd.Series([0, 0, 0, 1, 1, 1],
+    ...                  index=['s1','s2','s3','s4','s5','s6'])
+
+    Now run ancom and see if there are any features that have any
+    significant differences
+    >>> W, reject = ancom(table, cats)
+    >>> print(W)
+    b1    6
+    b2    6
+    b3    2
+    b4    2
+    b5    2
+    b6    2
+    b7    2
+    dtype: float64
+    >>> print(reject)
+    b1     True
+    b2     True
+    b3    False
+    b4    False
+    b5    False
+    b6    False
+    b7    False
+    dtype: bool
+    """
+    if len(mat) != len(cats):
+        raise ValueError('The number of samples in mat needs'
+                         'to be the same as the number of samples'
+                         'in cats')
+    mat = _check_composition(mat, ignore_zeros=False)
+    cats = pd.Series(cats)
+
+    mat = mat.sort_index()
+    cats = cats.sort_index()
+    labs = mat.columns
+
+    mat = np.atleast_2d(mat.values)
+    cats = np.array(cats.values)
+    n_samp, n_feat = mat.shape
+
+    _logratio_mat = _log_compare(mat, cats, func)
+    logratio_mat = _logratio_mat + _logratio_mat.T
+
+    # Multiple comparisons
+    if multicorr:
+        logratio_mat = np.apply_along_axis(_holm, 1, logratio_mat)
+
+    np.fill_diagonal(logratio_mat, 1)
+
+    W = (logratio_mat < alpha).sum(axis=1)
+    c_start = max(W) / n_feat
+    if c_start < theta:
+        reject = np.array([False] * len(W))
+        return pd.Series(W, index=labs), pd.Series(reject, index=labs)
+
+    cutoff = c_start - np.linspace(0.05, 0.25, 5)
+    dels = np.zeros(len(cutoff))
+    prop_cut = np.zeros(len(cutoff), dtype=np.float32)
+    for cut in range(len(cutoff)):
+        prop_cut[cut] = sum(W > n_feat*cutoff[cut]) / len(W)
+    for i in range(len(cutoff)-1):
+        dels[i] = abs(prop_cut[i] - prop_cut[i+1])
+
+    if (dels[1] < tau) and (dels[2] < tau) and (dels[3] < tau):
+        nu = cutoff[1]
+    elif (dels[1] >= tau) and (dels[2] < tau) and (dels[3] < tau):
+        nu = cutoff[2]
+    elif (dels[2] >= tau) and (dels[3] < tau) and (dels[4] < tau):
+        nu = cutoff[3]
+    else:
+        nu = cutoff[4]
+    reject = (W >= nu*n_feat)
+    return pd.Series(W, index=labs), pd.Series(reject, index=labs)
+
+
+def _check_composition(x, ignore_zeros=True):
+    """
+    Checks to make sure that composition meets the mininum criteria
+    Also casts composition into a pandas dataframe
+
+    Parameters
+    ----------
+    x : array_like or pd.DataFrame
+       Input composition matrix
+       where rows=samples and columns=features
+
```

Comment:
Can you ad the description of ignore_zeros? ... or maybe not, as this is a private function and guidelines with those are kinda loose :scream: